src/pages/permissions/actions/DeleteGroupModal.tsx

```diff
@@ -0,0 +1,115 @@
+import {
+  ConfirmationModal,
+  Input,
+  useNotify,
+} from "@canonical/react-components";
+import { useQueryClient } from "@tanstack/react-query";
+import { deleteGroup, deleteGroups } from "api/auth-groups";
+import { useToastNotification } from "context/toastNotificationProvider";
+import { ChangeEvent, FC, useState } from "react";
+import { LxdGroup } from "types/permissions";
+import { pluralize } from "util/instanceBulkActions";
+import { queryKeys } from "util/queryKeys";
+
+interface Props {
+  groups: LxdGroup[];
+  close: () => void;
+}
+
+const DeleteGroupModal: FC<Props> = ({ groups, close }) => {
+  const queryClient = useQueryClient();
+  const notify = useNotify();
+  const toastNotify = useToastNotification();
+  const [confirmInput, setConfirmInput] = useState("");
+  const [disableConfirm, setDisableConfirm] = useState(true);
+  const [submitting, setSubmitting] = useState(false);
+  const hasOneGroup = groups.length === 1;
+  const confirmText = "confirm-delete-group";
+
+  const handleConfirmInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.value === confirmText) {
+      setDisableConfirm(false);
+    } else {
+      setDisableConfirm(true);
+    }
+
+    setConfirmInput(e.target.value);
+  };
+
+  const handleDeleteGroups = () => {
+    setSubmitting(true);
+    const hasSingleGroup = groups.length === 1;
+    const mutationPromise = hasSingleGroup
+      ? deleteGroup(groups[0].name)
+      : deleteGroups(groups.map((group) => group.name));
+
+    const successMessage = hasSingleGroup
+      ? `Group ${groups[0].name} deleted.`
+      : `${groups.length} groups deleted.`;
+
+    mutationPromise
+      .then(() => {
+        void queryClient.invalidateQueries({
+          predicate: (query) => {
+            return [queryKeys.identities, queryKeys.authGroups].includes(
+              query.queryKey[0] as string,
+            );
+          },
+        });
+        toastNotify.success(successMessage);
+      })
+      .catch((e) => {
+        notify.failure(
+          `${pluralize("group", groups.length)} deletion failed`,
+          e,
+        );
+      })
+      .finally(() => {
+        setSubmitting(false);
+        close();
```

Comment:
I thought it would be better to close this modal for errors as well since there is nothing the user can do in this modal to investigate/fix the error?

Reply:
I thought they could retry, and it might be frustrating to have to type that confirmation text a 2nd time.